// driver-license-parser/build/main/lib/field_parsers/base_parser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseParser = void 0;
const license_1 = require("./license");
class BaseParser {
    constructor(map) {
        this.fieldMappers = new Map();
        this.fieldData = {};
        this.defaultMapperObject = map;
        Object.keys(this.defaultMapperObject).forEach((key) => {
            this.fieldMappers.set(key, map[key]);
        });
    }
    parse(pdf417Data) {
        const result = {};
        const lines = pdf417Data.split('\n');
        Object.keys(this.defaultMapperObject).forEach((key) => {
            const regex = new RegExp(`${this.defaultMapperObject[key]}(.+)`);
            lines.forEach((line) => {
                const match = regex.exec(line);
                if (match) {
                    const data = match[1];
                    this.fieldData[key] = data;
                    result[key] = data;
                }
            });
        });
        return this.generateLicense();
    }
    generateLicense() {
        const license = new license_1.License();
        license.firstName = this.getFirstName();
        license.lastName = this.getLastName();
        license.middleName = this.getMiddleName();
        license.dateOfBirth = this.getDateOfBirth();
        license.street = this.getStreetAddress();
        license.city = this.getCity();
        license.state = this.getState();
        license.zip = this.getZip();
        license.issueDate = this.getIssueDate();
        license.expirationDate = this.getExpirationDate();
        return license;
    }
    isState(key) {
        return key === 'state';
    }
    isDate(key) {
        return (key === 'dateOfBirth' || key === 'issueDate' || key === 'expirationDate');
    }
    parseString(str) {
        return str
            .split(' ')
            .map((a) => a.charAt(0).toUpperCase() + a.slice(1).toLowerCase())
            .join(' ');
    }
    parseDate(date) {
        const year = date.substring(4, 8);
        const month = date.substring(2, 4);
        const day = date.substring(0, 2);
        return `${day}/${month}/${year}`;
    }
    isGender(key) {
        return key === 'gender';
    }
    get(field) {
        return this.fieldData[field] || '';
    }
    getFirstName() {
        return this.parseString(this.get('firstName'));
    }
    getLastName() {
        return this.parseString(this.get('lastName'));
    }
    getMiddleName() {
        return this.parseString(this.get('middleName'));
    }
    getGender() {
        const gen = this.get('gender');
        return this.parseGender(gen);
    }
    parseGender(gen) {
        switch (gen) {
            case '1':
                return 'Male';
            case '2':
                return 'Female';
            default:
                return 'Unknown';
        }
    }
    getDateOfBirth() {
        return this.parseDate(this.get('dateOfBirth'));
    }
    getIssueDate() {
        return this.parseDate(this.get('issueDate'));
    }
    getExpirationDate() {
        return this.parseDate(this.get('expirationDate'));
    }
    getStreetAddress() {
        return this.parseString(this.get('street'));
    }
    getCity() {
        return this.parseString(this.get('city'));
    }
    getState() {
        return this.get('state');
    }
    getZip() {
        return this.get('zip');
    }
    getHairColor() {
        return this.parseHairColor(this.get('hairColor'));
    }
    eyeColor() {
        return this.get('eyeColor');
    }
    parseHairColor(color) {
        switch (color) {
            case 'BAL':
                return 'Bald';
            case 'BLK':
                return 'Black';
            case 'BLN':
                return 'Blond';
            case 'BRO':
                return 'Brown';
            case 'GRY':
                return 'Grey';
            case 'RED':
                return 'Red';
            case 'SDY':
                return 'Sandy';
            case 'WHI':
                return 'White';
            default:
                return 'Unknown';
        }
    }
}
exports.BaseParser = BaseParser;
