// driver-license-parser/build/main/lib/field_parsers/license.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.License = void 0;
class License {
    constructor(license = {}) {
        this.firstName = '';
        this.lastName = '';
        this.middleName = '';
        this.issueDate = '';
        this.expirationDate = '';
        this.dateOfBirth = '';
        this.street = '';
        this.secondStreet = '';
        this.city = '';
        this.state = '';
        this.zip = '';
        this.hairColor = '';
        this.firstName = license.firstName || '';
        this.lastName = license.lastName || '';
        this.middleName = license.middleName || '';
        this.issueDate = license.issueDate || '';
        this.expirationDate = license.expirationDate || '';
        this.dateOfBirth = license.dateOfBirth || '';
        this.street = license.street || '';
        this.secondStreet = license.secondStreet || '';
        this.city = license.city || '';
        this.state = license.state || '';
        this.zip = license.zip || '';
    }
    getFullName() {
        return `${this.firstName} ${this.lastName}`;
    }
    isExpired() {
        if (!this.expirationDate)
            return false;
        const now = new Date();
        const expirationDate = new Date(this.expirationDate);
        return now > expirationDate;
    }
    isUnder21() {
        if (!this.dateOfBirth)
            return false;
        const now = new Date();
        const dateOfBirth = new Date(this.dateOfBirth);
        let age = now.getFullYear() - dateOfBirth.getFullYear();
        const month = now.getMonth() - dateOfBirth.getMonth();
        if (month < 0 || (month === 0 && now.getDate() < dateOfBirth.getDate())) {
            age--;
        }
        return age < 21;
    }
    streetAddress() {
        return `${this.street.length ? this.street + ',' : ''} ${this.secondStreet}`;
    }
}
exports.License = License;
