// driver-license-parser/build/module/lib/field_parsers/version_one/parser.js
import { BaseParser } from "../base_parser";
import { FieldMap } from "./field_mapper";
export class VersionOneParser extends BaseParser {
    constructor() {
        super(Object.assign({}, FieldMap));
    }
    parseDriverLicenseName(key) {
        const names = this.get('driverLicenseName').split(',');
        switch (key) {
            case 'firstName':
                return names.length > 2 ? names[1] : '';
            case 'lastName':
                return names.length > 1 ? names[0] : '';
            case 'middleName':
                return names.length > 3 ? names[2] : '';
            case 'suffix':
                return names.length > 4 ? names[3] : '';
        }
        return '';
    }
    getFirstName() {
        return this.parseString(this.get('firstName'));
    }
    getLastName() {
        return this.parseString(this.get('lastName'));
    }
    getMiddleName() {
        return this.parseString(this.get('middleName'));
    }
    getSuffix() {
        return this.parseString(this.get('suffix'));
    }
    parseDate(date) {
        const year = date.substring(0, 4);
        const month = date.substring(4, 6);
        const day = date.substring(6, 8);
        return `${day}/${month}/${year}`;
    }
}
